src/components/pages/home/AddressLink.tsx

```diff
@@ -0,0 +1,52 @@
+import { useContext } from "react";
+import Link from "next/link";
+import { AddressContext } from "@/components/lib/context/AddressContext";
+
+interface AddressLinkProps {
+  type: string;
+  label: string;
+  address: string | null;
+  href: string;
+  children: React.ReactNode;
+}
+
+const AddressLink = ({
+  type,
+  label,
+  address,
+  href,
+  children,
+}: AddressLinkProps) => {
+  const { state, dispatch } = useContext(AddressContext);
+
+  const handleMouseEnter = () => {
+    dispatch({ type: "SET_HOVERED_ADDRESS", payload: address });
+  };
+
+  const handleMouseLeave = () => {
+    dispatch({ type: "SET_HOVERED_ADDRESS", payload: null });
+  };
+
+  const className = `truncate ${
+    type === "l1-l2"
+      ? "max-w-28 md:max-w-48 xl:max-w-28"
+      : type === "latest-transactions" && label === "tx"
+        ? "max-w-60 md:max-w-96 xl:max-w-60"
+        : "max-w-36 md:max-w-48 xl:max-w-36"
+  }`;
+
+  return (
+    <Link
+      href={href}
+      className={`flex items-center text-primary hover:brightness-150 ${
+        state.hoveredAddress === address ? "text-yellow-200" : ""
+      }`}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <span className={className}>{children}</span>
```

Comment:
We need to be able to display the external link icon here, see screenshot attached.

![Screenshot 2024-06-24 at 12 28 58](https://github.com/walnuthq/op-scan/assets/5597359/5c1c43cf-2585-4c96-94a3-cec7431feb66)

We should display the external link icon only if the href is pointing to an external URL (`new URL(href).origin !== location.origin`)